pkg/authorization/controller/authorizationsync/origin_to_rbac_clusterrole_controller.go

```diff
@@ -120,14 +121,26 @@ func (c *OriginClusterRoleToRBACClusterRoleController) clusterPolicyEventHandler
 			}
 		},
 		UpdateFunc: func(old, cur interface{}) {
-			originContainerObj := cur.(*authorizationapi.ClusterPolicy)
-			for _, originObj := range originContainerObj.Roles {
+			keys := sets.NewString()
+			for _, originObj := range old.(*authorizationapi.ClusterPolicy).Roles {
+				c.originIndexer.Delete(originObj)
```

Comment:
Seems like this would put our lister into a bad state that could be observed by the syncFunction, resulting in bad behavior as we repeatedly delete and recreate roles.